### ISSUE TITLE
kinfocenter: kcm_energyinfo library not found

### ISSUE THREAD

**Issue by wincak** [manjaro/packages-extra#138]
Kinfocenter plugin kcm_energyinfo is installed in /usr/lib/qt/plugins/kcms/kcm_energyinfo.so and kinfocenter cannot find it. This (probably) makes some kinfocenter functionality unavailable.

If kinfocenter is run from console, it shows errors:
org.kde.kcoreaddons: Error loading plugin "kcm_energyinfo" "The shared library was not found." 
Plugin search paths are ("/usr/lib/qt/plugins", "/usr/bin") 
The environment variable QT_PLUGIN_PATH might be not correctly set

And indeed the command
$ echo $QT_PLUGIN_PATH
doesn't return any path (just an empty string).

I have previously filed this bug with KDE as:
https://bugs.kde.org/show_bug.cgi?id=387366
but it has been rejected as a packaging issue.

I'm using the Manjaro i3 edition and have installed KDE from repositories. All packages are up-to-date.

**Comment by philmmanjaro** [comment]
**kinfocenter** is an [Arch package](https://www.archlinux.org/packages/extra/x86_64/kinfocenter/). `usr/lib/qt/plugins/kcms/kcm_energyinfo.so` is part of **kinfocenter** itself. Please check if **v5.11.4** fixes the issue. Else fill out a [bug report](https://bugs.archlinux.org/newtask?project=1&product_category=2&item_summary=%5Bkinfocenter%5D+PLEASE+ENTER+SUMMARY).

**Comment by oberon2007** [comment]
No, we package it ourselves. I just had a closer look and cannot find any problem with the package itself. The library **/usr/lib/qt/plugins/kcms/kcm_energyinfo.so** is certainly installed.
Will now update and see it that fixes it.

**Comment by philmmanjaro** [comment]
Currently updating it already.

**Comment by oberon2007** [comment]
Doesn't help.

**Comment by philmmanjaro** [comment]
Hmm, question is why all other plugins are not in a separate **kcms** folder:

```
usr/lib/qt/plugins/
usr/lib/qt/plugins/kcm_about_distro.so
usr/lib/qt/plugins/kcm_devinfo.so
usr/lib/qt/plugins/kcm_info.so
usr/lib/qt/plugins/kcm_memory.so
usr/lib/qt/plugins/kcm_nic.so
usr/lib/qt/plugins/kcm_opengl.so
usr/lib/qt/plugins/kcm_pci.so
usr/lib/qt/plugins/kcm_samba.so
usr/lib/qt/plugins/kcm_usb.so
usr/lib/qt/plugins/kcm_view1394.so
usr/lib/qt/plugins/kcms/
usr/lib/qt/plugins/kcms/kcm_energyinfo.so
usr/lib/qt/plugins/kcms/kcm_fileindexermonitor.so
```

**Comment by philmmanjaro** [comment]
See also: https://bugs.kde.org/show_bug.cgi?id=364637

**Comment by philmmanjaro** [comment]
@wincak: please check aslo file indexer monitor plugin with this changes made ...

**Comment by oberon2007** [comment]
Still error for both:
```
Error loading plugin: "The shared library was not found."
This module has no valid entry symbol at all. The reason could be that it's still using K_EXPORT_COMPONENT_FACTORY with a custom X-KDE-FactoryName which is not supported anymore
```

**Comment by philmmanjaro** [comment]
Then I don't know. Seems we have to reopen the upstream bug. We may check other packages from other distros ...

**Comment by philmmanjaro** [comment]
https://github.com/NixOS/nixpkgs/issues/27697

**Comment by oberon2007** [comment]
Yes, I agree. Also I found that other distributions have the same issue:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=860041
https://github.com/NixOS/nixpkgs/issues/27697
https://bugzilla.redhat.com/show_bug.cgi?id=1456058#c0

**Comment by oberon2007** [comment]
exactly.
It's a KDE issue.

**Comment by philmmanjaro** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=860041

**Comment by philmmanjaro** [comment]
https://bugs.kde.org/show_bug.cgi?id=376764

> Antonio Rojas 2017-02-21 16:21:26 UTC
> 
> Ah no, sorry. We fixed installing look and feel packages, but the error message is still there. It happens for all plugins installed in /usr/lib/qt/plugins/kcms, but doesn't seem to cause any actual issue.

**Comment by philmmanjaro** [comment]
I assume if we install the Arch-Package it is the same matter...

**Comment by oberon2007** [comment]
our PKGBUILD is practically identical.
And, it DOES cause issues. Those modules are not working 😉

**Comment by philmmanjaro** [comment]
Then we should involve Antonio Rojas from Archlinux into this matter ...

**Comment by oberon2007** [comment]
Yes. I've just verified with the original Arch pkg from their mirror. Same issue.

**Comment by philmmanjaro** [comment]
Triggered Arch: https://bugs.archlinux.org/task/56502

**Comment by wincak** [comment]
Wow, thanks for working on this issue so fast.
@philmmanjaro I'm still on v5.11.3-1 and can't see any available update...

kcm_fileindexermonitor behaves in a similar way
```
org.kde.kcoreaddons: Error loading plugin "kcm_fileindexermonitor" "The shared library was not found." 
Plugin search paths are ("/usr/lib/qt/plugins", "/usr/bin") 
The environment variable QT_PLUGIN_PATH might be not correctly set
```
After symlinking /usr/lib/qt/plugins/kcms/kcm_fileindexermonitor.so to /usr/lib/qt/plugins/kcm_fileindexermonitor.so the message disappears.
However kinfocenter's File Indexer Monitor tab seems to be working OK in both cases.

![screenshot_20171129_182745](https://user-images.githubusercontent.com/12111616/33386250-34737e42-d533-11e7-9682-3a71767b091f.png)

Also, if that is relevant, in total the /usr/lib/qt/plugins/kcms folder on my computer contains:
kcm_desktoptheme.so - belongs to plasma-desktop
kcm_fileindexermonitor.so - kinfocenter
kcm_pulseaudio.so - plasma-pa
kcm_energyinfo.so - kinfocenter
kcm_lookandfeel.so - plasma-desktop
kcm_splashscreen.so - plasma-desktop
But most of kcm_* libraries are in the parent folder.

And this is what the energy info tab looks like. It actually is displaying something, just the charge and consumption statistics are missing. So maybe kinfocenter complains but finds these libraries in the end and the missing functionality is an unrelated issue?
![screenshot_20171129_182723](https://user-images.githubusercontent.com/12111616/33386167-08a54bec-d533-11e7-82a5-21b3c114aba2.png)

**Comment by philmmanjaro** [comment]
When the KCM displays all is fine. Please see also the upstream bug replies:

https://bugs.archlinux.org/task/56502
https://bugs.kde.org/show_bug.cgi?id=387366

**Comment by oberon2007** [comment]
For me Energy Information doesn't display anything. Same with the File Indexer.
Manual symlinking should in any case not be necessary.
